### PR TITLE
#5384 Google Sheets slice 1b + 3 - Switch to supporting variables (mod inputs) in GSheets bricks

### LINKED ISSUE #5398: I.3 Deprecate Google Sheets as “Integrations”
<https://www.notion.so/pixiebrix/Make-Google-Sheets-work-the-way-users-expect-041e91ede0d04a928d21b7bfe7f3482f#c9af83b281964bc0a5a07bb55bc41e8e>

Phase I Slice 3

<p></p>

Note: The deprecation part is likely going to end up being covered as part of <https://github.com/pixiebrix/pixiebrix-extension/issues/5384>  
I'm just opening this issue for tracking purposes\.

<p></p>

However, we do still need to remove the feature flag before release, so we can use this issue for that\.

<p></p>

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #5384 
- Closes #5398 

## Demo

https://www.loom.com/share/2f1dd1e368d742d3a6b688e793132e3d

#### Slice 1
- ~Put the @variable field toggle option in the page editor behind the feature flag~
- ~Improve useSpreadSheetId test coverage to run all cases with feature flag on and off~
- Add logic that attempts to clean up GSheets integration from `services` on the extension when a GSheets brick spreadsheetId field is toggled from @variable to the Sheet Picker, if the service isn't used anywhere else
- ~UI tests for Append/Lookup bricks~
#### Slice 3
- Remove GSheets as a service integration option in the Extension & Admin Consoles

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer

### CODE REVIEW

File: src/testUtils/testHelpers.tsx [hunk 1]
```diff
@@ -301,6 +306,11 @@ export function createRenderHookWithWrappers(configureStore: ConfigureStore) {
         return store;
       },
       act: actHook,
+      async waitForEffect() {
```
Comment (twschiller) [inline src/testUtils/testHelpers.tsx:309]: 👍

File: src/contrib/google/sheets/useSpreadsheetId.test.ts [hunk 2]
```diff
@@ -45,34 +65,116 @@ describe("useSpreadsheetId", () => {
...
+  );
+
+  test.each(cases)(
+    "works with service value (flag gsheets-mod-inputs $flagOn)",
+    async ({ setupFlag }) => {
+      setupFlag();
+
+      const { result, waitForEffect } = renderHook(() => useSpreadsheetId(""), {
+        initialValues: {
+          spreadsheetId: {
```
Comment (twschiller) [inline src/contrib/google/sheets/useSpreadsheetId.test.ts]: Can use `makeVarExpression` helper

File: src/components/fields/schemaFields/getToggleOptions.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faCloud } from "@fortawesome/free-solid-svg-icons";
 import { faFileAlt } from "@fortawesome/free-regular-svg-icons";
 import { ServiceFieldDescription } from "@/components/fields/schemaFields/ServiceField";
+import { addListener } from "@/auth/token";
```
Comment (twschiller) [inline src/components/fields/schemaFields/getToggleOptions.tsx:27]: Rename the import to be more meaningful, e.g., 
```
import { addListener: addAuthListener } from "@/auth/token";
```

File: src/components/fields/schemaFields/getToggleOptions.tsx [hunk 4]
```diff
@@ -34,6 +35,12 @@ type ToggleOptionInputs = {
   allowExpressions: boolean;
 };
 
+// TODO: extract/abstract this pattern if we need to synchronously access flags anywhere else
+const localFeatureFlags: string[] = [];
+addListener(({ flags }) => {
```
Comment (twschiller) [inline src/components/fields/schemaFields/getToggleOptions.tsx:40]: I don't think addListener is called on module load. Are you sure this actually runs?

File: src/contrib/google/sheets/useSpreadsheetId.ts [hunk 5]
```diff
@@ -97,8 +100,26 @@ function useSpreadsheetId(basePath: string): string | null {
...
   useAsyncEffect(async () => {
     dispatch(spreadsheetSlice.actions.setLoading());
-    if (isServiceValueFormat(fieldValue)) {
-      if (fieldValue == null) {
+    if (
+      isVarExpression(fieldValue) &&
+      startsWith(fieldValue.__value__, "@options.") &&
+      !isEmpty(optionsArgs)
+    ) {
+      const optionsKey = fieldValue.__value__.replace("@options.", "");
```
Comment (twschiller) [inline src/contrib/google/sheets/useSpreadsheetId.ts:108]: I'd recommend extracting out the "getOptionsArgFromVariable" as a method

File: src/components/fields/schemaFields/getToggleOptions.tsx (hunk 4 above)
Comment (twschiller) [reply src/components/fields/schemaFields/getToggleOptions.tsx:40]: I recommend making a separate module called syncFlags for the synchronous flag logic in the "store" folder

Comment (BLoe) [reply src/components/fields/schemaFields/getToggleOptions.tsx:40]: Yeah this doesn't initialize the flags on load at all 🤦

File: src/contrib/google/sheets/useSpreadsheetId.ts (hunk 5 above)
Comment (twschiller) [reply src/contrib/google/sheets/useSpreadsheetId.ts:108]: > Do you think we'll use it again in the future in other places?

Splitting it out makes unit testing the logic easier

**Pull request merged**